<a href="https://colab.research.google.com/github/chw8207/python-seminar/blob/main/Pandas_%EC%97%B0%EC%8A%B5_%EB%AC%B8%EC%A0%9C.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import pandas as pd

In [ ]:
sample = pd.read_csv("/content/drive/MyDrive/파이썬 세미나/세미나 내용/python-seminar/data/내국인 생활인구 - 내국인 생활인구.csv")
sample.head()

일자(stdr_de_id)  시간(tmzon_pd_se)  행정동코드(adstrd_code_se)   집계구코드(oa_cd)  \
0        20170226                4               11320515  1125061020007   
1        20180107               22               11380510  1115071010309   
2        20180317                6               11680610  1113073020101   
3        20190222                7               11305575  1106071020403   
4        20210706                6               11740660  1107073040015   

   생활인구합계(tot_lvpop_co)  남자0~9세(male_f0t9_lvpop_co)  \
0                4.3097                     11.9781   
1              338.3900                     23.2446   
2             1805.5256                      4.3720   
3              504.4956                     14.0599   
4              744.6219                      0.5662   

   남자10~14세(male_f10t14_lvpop_co)  남자15~19세(male_f15t19_lvpop_co)  \
0                          8.2257                          1.8759   
1                          3.1141                          1.9972   
2                        324.3596                         10.8350   
3                         14.3256                          3.5239   
4                          2.9824                          2.9719   

   남자20~24세(male_f20t24_lvpop_co)  남자25~29세(male_f25t29_lvpop_co)  ...  \
0                          4.2075                          6.2647  ...   
1                          2.6983                          0.1240  ...   
2                         12.1324                         20.4938  ...   
3                         19.3154                          6.8197  ...   
4                         31.0868                          7.2338  ...   

   여자40~44세(female_f40t44_lvpop_co)  여자45~49세(female_f45t49_lvpop_co)  \
0                            9.0816                           52.6711   
1                           10.0707                            9.9150   
2                           28.8794                           19.4895   
3                            0.2650                           36.5731   
4                           39.7471                            1.5230   

   여자50~54세(female_f50t54_lvpop_co)  여자55~59세(female_f55t59_lvpop_co)  \
0                           24.9152                            3.6202   
1                           14.6710                           29.2192   
2                            1.6087                           16.2525   
3                           14.3217                           35.7359   
4                           47.5514                           63.4768   

   여자60~64세(female_f60t64_lvpop_co)  여자65~69세(female_f65t69_lvpop_co)  \
0                           91.3448                           13.6818   
1                            0.1533                           36.7707   
2                            8.1640                           34.3989   
3                            7.2808                            6.9919   
4                           52.3638                           35.3496   

   여자70~74세(female_f70t74_lvpop_co)  여자75~79세(female_f75t79_lvpop_co)  \
0                           45.2880                            2.3141   
1                            0.9227                            4.6111   
2                            3.4030                            3.8434   
3                           27.4321                            1.8361   
4                           32.1275                           85.6665   

   여자80세이상(female_above_80_lvpop_co)      등록일자(ldadng_dt)  
0                             9.7668  2021-08-17 16:06:44  
1                            18.1781  2021-08-17 16:06:44  
2                             4.4772  2021-08-17 16:06:44  
3                             7.7510  2021-08-17 16:06:44  
4                            95.0249  2021-08-17 16:06:44  

[5 rows x 38 columns]

## 1. 컬럼명 바꾸기

sample에 있는 컬럼명에서 한글 컬럼명만 남겨주세요.  
예시: `일자(stdr_de_id)` -> `일자`

In [16]:
# 컬럼명 조회
sample.columns

Index(['일자(stdr_de_id)', '시간(tmzon_pd_se)', '행정동코드(adstrd_code_se)',
       '집계구코드(oa_cd)', '생활인구합계(tot_lvpop_co)', '남자0~9세(male_f0t9_lvpop_co)',
       '남자10~14세(male_f10t14_lvpop_co)', '남자15~19세(male_f15t19_lvpop_co)',
       '남자20~24세(male_f20t24_lvpop_co)', '남자25~29세(male_f25t29_lvpop_co)',
       '남자30~34세(male_f30t34_lvpop_co)', '남자35~39세(male_f35t39_lvpop_co)',
       '남자40~44세(male_f40t44_lvpop_co)', '남자45~49세(male_f45t49_lvpop_co)',
       '남자50~54세(male_f50t54_lvpop_co)', '남자55~59세(male_f55t59_lvpop_co)',
       '남자60~64세(male_f60t64_lvpop_co)', '남자65~69세(male_f65t69_lvpop_co)',
       '남자70~74세(male_f70t74_lvpop_co)', '남자75~79세(male_f75t79_lvpop_co)',
       '남자80세이상(male_above_80_lvpop_co)', '여자0~9세(female_f0t9_lvpop_co)',
       '여자10~14세(female_f10t14_lvpop_co)', '여자15~19세(female_f15t19_lvpop_co)',
       '여자20~24세(female_f20t24_lvpop_co)', '여자25~29세(female_f25t29_lvpop_co)',
       '여자30~34세(female_f30t34_lvpop_co)', '여자35~39세(female_f35t39_lvpop_co)',
       '여자40~44세(f

## 2. 인덱스 설정
`일자` 컬럼을 인덱스로 설정해주세요.

일자 인덱스를 datetime 형식으로 변경한 뒤, 날짜 오름차 순으로 데이터를 정렬할 것입니다.  
astype() 메서드를 사용해서 sample.index를 str형으로 바꿔주세요.

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.astype.html

str로 변경된 인덱스를 pd.to_datetime 메서드를 사용하여 datetime 형식으로 바꿔주세요.

https://pandas.pydata.org/docs/reference/api/pandas.to_datetime.html

sort_index() 메서드를 사용하여 주어진 데이터를 인덱스 날짜 오름차 순으로 정렬해주세요.

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sort_index.html

## 3. 조건에 맞는 데이터 추출
집계구 코드가 `1107073040015` 또는 `1113073020101`인 행을 뽑아주세요.

## 4. apply 활용

각 행에 대해 남성 인구의 합을 구하여 `남자인구 합계` 이름을 가진 컬럼을 생성해주세요.

## 5. 시각화

남성인구 중 남자0\~9세가 차지하는 비율을 계산하여 "남자0~9세 비율" 컬럼을 만들어주세요.

시간의 흐름에 따라 전체 남자 생활인구 대비 `남자0~9세 비율`의 변화를 한 눈에 알 수 있도록  
`남자0~9세 비율`을 선 그래프로 나타내주세요.


* 한글 폰트가 깨지지 않도록 처리해주세요.
* 그래프 사이즈는 20\*10으로 해주세요.